RecoEgamma/EgammaTools/interface/ShowerDepth.h

```diff
@@ -0,0 +1,57 @@
+#ifndef RecoEgamma_EgammaTools_ShowerDepth_h
+#define RecoEgamma_EgammaTools_ShowerDepth_h
+
+/* ShowerDepth: computes expected EM shower mean depth
+ * in the HGCal, and compares it to measured depth
+ *
+ * Code copied from C. Charlot's
+ * Hopefully correct explanations by N. Smith
+ *
+ * Based on gamma distribution description of electromagnetic cascades [PDG2016, ss33.5]
+ * Basic equation:
+ *   D_exp = X_0 t_max alpha / (alpha-1)
+ * where D_exp is expectedDepth, X_0 the radiation length in HGCal material, t_max
+ * the shower maximum and alpha a parameter of the gamma distribution. (beta := (a-1)/t_max)
+ * Both t_max and alpha are approximated as first order polynomials of ln y = E/E_c, where E_c
+ * is the critical energy, and presumably are extracted from fits to simulation, along with their
+ * corresponding uncertainties.
+ *
+ * sigma(D_exp) then follows from error propagation and we can compare to
+ * measured depth (distance between first hit and shower barycenter)
+ *
+ * The parameterization is for electrons, although photons will give similar results
+ */
+
+class ShowerDepth {
```

Comment:
Following on the comment made in https://github.com/cms-sw/cmssw/pull/21037#pullrequestreview-73117488
 If this can work only for HGC, I think all of the new code better reflect it in the name.
HGCShowerDepth will help to avoid possible symbol collisions.
A simpler solution is to use ``` namespace hgcal ```

Reply:
Good point about collisions, I'll wrap these smaller classes with a namespace.